src/clib/lib/enkf/obs_vector.cpp

```diff
@@ -53,6 +54,22 @@ struct obs_vector_struct {
     std::vector<int> step_list;
 };
 
+static int find_within_tolerance(std::vector<time_t> time_map,
+                                 time_t external_time) {
+    int nearest_index = -1;
+    auto nearest_diff = std::numeric_limits<double>::infinity();
+    for (auto &&[index, time] : iter::enumerate(time_map)) {
+        time_t diff = std::abs(external_time - time);
+        if (diff == 0)
+            return index;
+        if (diff < 30 && diff < nearest_diff) {
```

Comment:
Why is the tolerance set to 30 seconds?

Reply:
No good reason, it was always like that. It was input to the original function:
```c++
int time_map_lookup_time_with_tolerance(time_map_type *map, time_t time,
                                        int seconds_before_tolerance,
                                        int seconds_after_tolerance) {
    int nearest_index = -1;
    pthread_rwlock_rdlock(&map->rw_lock);
    {
        if (time_map_valid_time__(map, time)) {
            time_t nearest_diff = 999999999999;
            int current_index = 0;
            while (true) {
                time_t diff = time - time_map_iget__(map, current_index);
                if (diff == 0) {
                    nearest_index = current_index;
                    break;
                }

                if (std::fabs(diff) < nearest_diff) {
                    bool inside_tolerance = true;
                    if (seconds_after_tolerance >= 0) {
                        if (diff >= seconds_after_tolerance)
                            inside_tolerance = false;
                    }

                    if (seconds_before_tolerance >= 0) {
                        if (diff <= -seconds_before_tolerance)
                            inside_tolerance = false;
                    }

                    if (inside_tolerance) {
                        nearest_diff = diff;
                        nearest_index = current_index;
                    }
                }

                current_index++;

                if (current_index >= time_t_vector_size(map->map))
                    break;
            }
        }
    }
    pthread_rwlock_unlock(&map->rw_lock);
    return nearest_index;
```
but it was hard coded to 30 seconds

Reply:
Add a comment along the lines of "The tolerance of 30s is arbitrarily chosen."?